src/service/resources-impl.js

```diff
@@ -866,12 +869,23 @@ export class Resources {
         const resource = request.resource;
         const box = resource.getLayoutBox();
         const iniBox = resource.getInitialLayoutBox();
-        const diff = request.newHeight - box.height;
+        const margins = resource.getLayoutMargins();
+        const topMarginDiff = request.newMargins ?
+            request.newMargins.top - margins.top : 0;
+        const bottomMarginDiff = request.newMargins ?
+            request.newMargins.bottom - margins.bottom : 0;
+        const heightDiff = request.newHeight - box.height;
+        // The lowest boundary of the element that would appear to be resized
+        // as a result of this size change. If the bottom margin is being
+        // changed then it is the bottom edge of the margin box, otherwise it
+        // is the bottom edge of the layout box.
+        const bottomDisplacedBoundary = bottomMarginDiff ?
+            box.bottom + margins.bottom : box.bottom;
```

Comment:
I'm not sure what your suggestion is here?
The point in this is to record the boundary that will have a visible change as far as the user is concerned. If there is no margin change then it will only appear as though the bottom of the border-box has moved (which is box.bottom). If the bottom margin is also being resized then the spacing between this element and the next will also change.

So for example, if there is an element where box.bottom is <10% inside the top of the viewport and it has a bottom margin of 100px, then a pure height resize is fine (and would be allowed under the existing logic, as well as this new logic). If the bottom margin, however, is being resized then it's not fine, since this will make things within the main 80% of the viewport move around.